pil-analyzer/src/pil_analyzer.rs

```diff
@@ -348,6 +348,26 @@ impl PILAnalyzer {
                 vec![]
             }
             PilStatement::Include(_, _) => unreachable!(),
+            PilStatement::TraitDeclaration(_, _) => {
+                let names = statement
+                    .symbol_definition_names_and_contained()
+                    .filter_map(|(name, sub_name, symbol_category)| match sub_name {
+                        None => Some((self.driver().resolve_decl(name), symbol_category)),
+                        Some(_) => None,
```

Comment:
Why are you filtering out the trait functions here (i.e. why are you not using the generic impl below)? Again: If it's not done for enums, why should it be done for trait functions?

Reply:
I think we shouldn't even make the implementation visible. You will always call the trait function and you will specify the type params. It is up to the evaluator to find the correct implementation (and of course the type checker).

Also there will be multiple implementations, if you add them to the namespace, they will clash.

Reply:
Perfect. I had a solution in my other PR to avoid collisions between implementations, but surely now it will be solved in a cleaner way. Thanks and sorry for the time lost.